src/Microsoft.AspNetCore.SignalR.Specification.Tests/ScaleoutHubLifetimeManagerTests.cs

```diff
@@ -0,0 +1,412 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.SignalR.Protocol;
+using Microsoft.AspNetCore.SignalR.Tests;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.SignalR.Specification.Tests
+{
+    public abstract class ScaleoutHubLifetimeManagerTests : HubLifetimeManagerTestsBase<MyHub>
+    {
+        public abstract HubLifetimeManager<MyHub> CreateNewHubLifetimeManagerFromExistingServer();
```

Comment:
If we break this up a little, I think we can have a cleaner API. I think we should split this into an API for creating a "Backplane" (i.e. TestRedisServer in the case of the Redis one). Then for our tests that want to test two "servers" connected via a common backplane, the test would call this API, then pass the Backplane object in to each `CreateHubLifetimeManager` method.

Something like this:

```csharp
public async Task TestMultipleServersConnectedToTheSameBackplane()
{
    var backplane = CreateBackplane();
    var server1 = CreateHubLifetimeManager(backplane);
    var server2 = CreateHubLifetimeManager(backplane);
    // The test...
}

public async Task TestOneServer()
{
    var backplane = CreateBackplane();
    var server = CreateHubLifetimeManager(backplane);
    // The test...
}
```

This also allows the test code to get an explicit call to initialize however they want.